src/tests/fuzz/fuzz_decodebase64.c

```diff
@@ -0,0 +1,55 @@
+/**
+ * @file
+ * @author Shivani Bhardwaj <shivani@oisf.net>
+ * fuzz target for DecodeBase64
+ */
+
+#include "suricata-common.h"
+#include "suricata.h"
+#include "util-base64.h"
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size);
+
+static int initialized = 0;
+
+static void ConstrainedDestinationSizeTest(const uint8_t *src, size_t len)
+{
+    for (uint8_t mode = BASE64_MODE_RELAX; mode <= BASE64_MODE_RFC4648; mode++) {
+        uint32_t dest_size = 3 * len / 4;
+        uint8_t *dest = malloc(dest_size);
+        uint32_t consumed_bytes = 0;
+        uint32_t decoded_bytes = 0;
+
+        DecodeBase64(dest, dest_size, src, len, &consumed_bytes, &decoded_bytes, mode);
+    }
+}
+
+static void AmpleDestinationSizeTest(const uint8_t *src, size_t len)
```

Comment:
Why do we need `AmpleDestinationSizeTest` ? Is `ConstrainedDestinationSizeTest` not enough ?

Reply:
I thought that if due to constrained size if some of the data wasn't fully decoded but was capable of hitting a Debug validation then this would help. Wdyt?

Reply:
One example is the case you pointed out where we don't have enough dest space for decoding 1byte. While that should be checked as that may happen, it should also be checked that if there is just 1 byte and enough destination space, it actually does get decoded properly.

Reply:
But `ConstrainedDestinationSizeTest` should always have enough space, right ?

Reply:
destination size is controlled by the caller and may not have enough space, I think.
refs:
1. https://github.com/OISF/suricata/blob/master/src/util-decode-mime.c#L1225
2. https://github.com/OISF/suricata/blob/master/src/util-base64.c#L178

Reply:
wdyt?

Reply:
I guess I would go for only one test, where the destination size is fuzzingly randomized

Reply:
That sounds a lot better. Thank you!